File: applicatie/src/main/java/com/vdzon/administratie/data/DataResource.java
package com.vdzon.administratie.data;

import com.vdzon.administratie.util.JsonUtil;
import spark.Spark;

import javax.inject.Inject;

public class DataResource {

    @Inject
    public DataResource(DataService dataService) {
        Spark.get("/load", dataService::loadData, JsonUtil.json());
    }
}

File: applicatie/src/main/java/com/vdzon/administratie/data/DataResource.kt
package com.vdzon.administratie.data

import com.vdzon.administratie.util.JsonUtil
import spark.Route
import spark.Spark

import javax.inject.Inject

class DataResource
@Inject
constructor(dataService: DataService) {
    init {
        Spark.get("/load", Route { req, res -> dataService.loadData(req, res) }, JsonUtil.json())
    }
}
